contrib/babelfishpg_tds/src/include/tds_request.h

```diff
@@ -269,6 +288,104 @@ typedef TDSRequestData *TDSRequest;
 #define TVP_COLUMN_ORDERING_TOKEN		0x11
 #define TVP_END_TOKEN				0x00
 
+/* Find namespace oid of a procedure based on proc name. */
+static inline Oid
+get_proc_namespace_oid(char **proc_name, char *curr_db)
+{
+	char *logical_sch_name;
+	char *physical_sch_name;
+	char *db_name;
+	char *schema_name;
+	char	  **splited_object_name;
+	Oid obj_schema_oid = InvalidOid;
+
+	/* resolve the three part name */
+	splited_object_name = pltsql_plugin_handler_ptr->split_object_name(*proc_name);
+	db_name = splited_object_name[1];
+	schema_name = splited_object_name[2];
+	*proc_name = splited_object_name[3];
+
+	if (!strcmp(db_name, ""))
+		db_name = curr_db;
+
+	if (!strcmp(schema_name, ""))
+	{
+		/* Find the default schema for current user and get physical schema name */
+		const char *user = pltsql_plugin_handler_ptr->pltsql_get_user_for_database(db_name);
+		schema_name = pltsql_plugin_handler_ptr->get_authid_user_ext_schema_name((const char *) db_name, user);
+	}
+
+	logical_sch_name = downcase_truncate_identifier(schema_name,strlen(schema_name), true);
+	/* Get physical schema name from logical schema name */
+	physical_sch_name = pltsql_plugin_handler_ptr->get_physical_schema_name(db_name, logical_sch_name);
+	/* Get namespace oid from physical schema name */
+	obj_schema_oid = get_namespace_oid(physical_sch_name, false);
+
+	return obj_schema_oid;
+
+}
+
+static inline Oid
+tds_get_proargtypes_oid(char *proname, Oid pronamespace, Oid user_id, char *targeted_arg_name)
+{
+	HeapTuple    tuple;
+	CatCList   *catlist;
+	Oid matched_type = InvalidOid;
+	char *hash;
+	char new_target_name[NAMEDATALEN];
+
+	if (strlen(targeted_arg_name) >= NAMEDATALEN)
```

Comment:
truncate identifier does this, we need to reuse the utility functions rather than implementing by self every time